### PR TITLE
added project_password as a new field into project model,added approp…

### PR DESCRIPTION
**Pull request by ishangujarathi**
…riate migrations and developed 2 API endpoints to set the password for the project and verify password for the project

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests / checks that you performed to verify your changes. Provide instructions so we can reproduce. Please delete options that are not relevant.

- [ ] Tested the API endpoint on Swagger with the 'Try it out' feature
- [ ] Tested the Models using the Django Shell
- [ ] Any other tests you performed for checking edge cases.
- [ ] Any other tests you performed for checking exceptions.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

## Checklist for Models

If you added new models / made changes to exsting models, please fill this checklist. 

- [ ] The primary key of the Model is an AutoField
- [ ] I have defined the `__str__` method for the Model
- [ ] If the field is optional, it has been spcified explicitly
- [ ] I have added a `verbose_name` for each Field.
- [ ] I have added a docstring for the Model
- [ ] The model is registered on Admin Site.
- [ ] I have pushed the migrations.

## Checklist for API

If you added new api endpoints / made changes to exsting endpoints, please fill this checklist. 

- [ ] The endpoint is accessible through Swagger.
- [ ] The endpoint supports permissions and authentication for different roles.
- [ ] All exceptions have been handled and appropriate status code is returned to the user.
- [ ] I have added docstrings for ViewSets / Serializers.

### CODE REVIEW

File: backend/projects/views.py [hunk 1]
```diff
@@ -4225,3 +4225,74 @@ def change_task_language_field_to_project_target_language(self, request, pk):
...
+
+    @action(
+        detail=True,
+        methods=["POST"],
+        url_name="set_password",
+    )
+    def set_password(self, request, pk=None):
+        try:
+            project = Project.objects.get(pk=pk)
+            password = request.data.get("password")
```
Comment (KunalTiwary) [inline backend/projects/views.py:4237]: There can be a case where password is absent in request.data, please handle that

File: backend/projects/views.py [hunk 2]
```diff
@@ -4225,3 +4225,74 @@ def change_task_language_field_to_project_target_language(self, request, pk):
...
+
+            if not password:
+                return Response(
+                    {"error": "Password not provided"},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+
+            project.set_project_password(password)
```
Comment (KunalTiwary) [inline backend/projects/views.py:4245]: Please add a try except block here

**Review by KunalTiwary — APPROVED** [review]

**Review by KunalTiwary — APPROVED** [review]
